kube/services/jobs/fence-visa-update-job.yaml

```diff
@@ -0,0 +1,109 @@
+        - name: fence-google-storage-creds-secret-volume
+          secret:
+            secretName: "fence-google-storage-creds-secret"
+        - name: shared-data
+          emptyDir: {}
+      containers:
+      - name: fence
+        GEN3_FENCE_IMAGE
+        imagePullPolicy: Always
+        env:
+            - name: PYTHONPATH
+              value: /var/www/fence
+            - name: CONCURRENCY
+              GEN3_CONCURRENCY|-value: ""-|
+            - name: CHUNK_SIZE
+              GEN3_CHUNK_SIZE|-value: ""-|
+            - name: THREAD_POOL_SIZE
+              GEN3_THREAD_POOL_SIZE|-value: ""-|
+            - name: BUFFER_SIZE
+              GEN3_BUFFER_SIZE|-value: ""-|
+            - name: slackWebHook
+              valueFrom:
+                  configMapKeyRef:
+                    name: global
+                    key: slack_webhook
+                    optional: true
+            - name: gen3Env
+              valueFrom:
+                  configMapKeyRef:
+                    name: global
+                    key: environment
+            - name: FENCE_PUBLIC_CONFIG
+              valueFrom:
+                configMapKeyRef:
+                  name: manifest-fence
+                  key: fence-config-public.yaml
+                  optional: true
+        volumeMounts:
+          - name: "yaml-merge"
+            readOnly: true
+            mountPath: "/var/www/fence/yaml_merge.py"
+            subPath: yaml_merge.py
+          - name: "config-volume"
+            readOnly: true
+            mountPath: "/var/www/fence/fence-config-secret.yaml"
+            subPath: fence-config.yaml
+          - name: "fence-google-app-creds-secret-volume"
+            readOnly: true
+            mountPath: "/var/www/fence/fence_google_app_creds_secret.json"
+            subPath: fence_google_app_creds_secret.json
+          - name: "fence-google-storage-creds-secret-volume"
+            readOnly: true
+            mountPath: "/var/www/fence/fence_google_storage_creds_secret.json"
+            subPath: fence_google_storage_creds_secret.json
+        command: ["/bin/bash" ]
+        args:
+          - "-c"
+          - |
+            echo "${FENCE_PUBLIC_CONFIG:-""}" > "/var/www/fence/fence-config-public.yaml"
+            python /var/www/fence/yaml_merge.py /var/www/fence/fence-config-public.yaml /var/www/fence/fence-config-secret.yaml > /var/www/fence/fence-config.yaml
+
+            create_command="fence-create update-visas"
+
+            if [[ -z "$CHUNK_SIZE" ]]; then
+                create_command+=" --chunk-size $CHUNK_SIZE"
+            fi
+            if [[ -z "THREAD_POOL_SIZE" ]]; then
+                create_command+=" --thread-pool-size $THREAD_POOL_SIZE"
+            fi
+            if [[ -z "BUFFER_SIZE" ]]; then
+                create_command+=" --buffer-size $BUFFER_SIZE"
+            fi
+            if [[ -z "CONCURRENCY" ]]; then
+                create_command+=" --concurrency $CONCURRENCY"
+            fi
```

Comment:
Should these be `if [[ -n ...` as opposed to `if [[ -z ...`?

When I tested `gen3 job run fence-visa-update`, the logs showed:
```
usage: fence-create update-visas [-h] [--chunk-size CHUNK_SIZE]
                                 [--concurrency CONCURRENCY]
                                 [--thread-pool-size THREAD_POOL_SIZE]
                                 [--buffer-size BUFFER_SIZE]
fence-create update-visas: error: argument --chunk-size: expected one argument
```
I think the command being executed is `fence-create update-visas --chunk-size --thread-pool-size --buffer-size --concurrency`.

Reply:
Shoot yeah you're right. I copied this from the cronjob so I think both are broken. I'll have to update both and then I'll retest